WD-Firmware/comrv/comrv_hooks.c

```diff
@@ -0,0 +1,139 @@
+/* 
+* SPDX-License-Identifier: Apache-2.0
+* Copyright 2019 Western Digital Corporation or its affiliates.
+* 
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+* 
+* http:*www.apache.org/licenses/LICENSE-2.0
+* 
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+/**
+* @file   comrv_hooks.c
+* @author Ronen Haen
+* @date   20.10.2019
+* @brief  The file implements the COM-RV hooks interfaces
+* 
+*/
+
+/**
+* include files
+*/
+#include "common_types.h"
+#include "string.h"
+#include "comrv_api.h"
+/**
+* definitions
+*/
+
+/**
+* macros
+*/
+
+/**
+* types
+*/
+
+/**
+* globals
+*/
+extern void* __OVERLAY_GROUPS_START__;
+
+#ifdef D_COMRV_FW_INSTRUMENTATION
+comrvInstrumentationArgs_t g_stInstArgs;
+#endif /* D_COMRV_FW_INSTRUMENTATION */
+
+
+/**
+* memory copy hook
+*
+* @param  none
+*
+* @return none
+*/
+void comrvMemcpyHook(void* pDest, void* pSrc, u32_t sizeInBytes)
+{
+   u32_t loopCount = sizeInBytes/(sizeof(u32_t)), i;
+   /* copy dwords */
+   for (i = 0; i < loopCount ; i++)
+   {
+      *((u32_t*)pDest + i) = *((u32_t*)pSrc + i);
+   }
+   loopCount = sizeInBytes - (loopCount*(sizeof(u32_t)));
+   /* copy bytes */
+   for (i = (i-1)*(sizeof(u32_t)) ; i < loopCount ; i++)
+   {
+      *((u08_t*)pDest + i) = *((u08_t*)pSrc + i);
+   }
+}
+
+/**
+* load overlay group hook
+*
+* @param
+*
+* @return none
+*/
+void* comrvLoadOvlayGroupHook(comrvLoadArgs_t* pLoadArgs)
+{
+   comrvMemcpyHook(pLoadArgs->pDest, (u08_t*)&__OVERLAY_GROUPS_START__ + pLoadArgs->uiGroupOffset, pLoadArgs->uiSizeInBytes);
+   return pLoadArgs->pDest;
+}
+
+/**
+* notification hook
+*
+* @param  pInstArgs - pointer to instrumentation arguments
+*
+* @return none
+*/
+#ifdef D_COMRV_FW_INSTRUMENTATION
+void comrvInstrumentationHook(const comrvInstrumentationArgs_t* pInstArgs)
+{
+   g_stInstArgs = *pInstArgs;
+}
+#endif /* D_COMRV_FW_INSTRUMENTATION */
+
+/**
+* error hook
+*
+* @param  pErrorArgs - pointer to error arguments
+*
+* @return none
+*/
+void comrvErrorHook(const comrvErrorArgs_t* pErrorArgs)
+{
+   /* we can't continue so loop forever */
+   while (1);
+}
+
+/**
+* crc calculation hook
+*
+* @param
+*
+* @return calculated CRC
+*/
+u32_t comrvCrcCalcHook(void* pAddress, u16_t memSizeInBytes)
```

Comment:
wrap comrvCrcCalcHook function with compile switch like function comrvInstrumentationHook

Reply:
the intention is to wrap the call to this function - there is already a CR comment for this in comrv.c